salvo/src/lib/source_manager.py

```diff
@@ -1,25 +1,47 @@
 """
-This module abstracts the higher level functions of managing source
-code
+This module abstracts the higher level functions of managing source code
 """
 import logging
 from typing import List
 
 from src.lib import source_tree
+
 import api.source_pb2 as proto_source
 import api.control_pb2 as proto_control
 
 log = logging.getLogger(__name__)
 
-"""The KNOWN_REPOSITORIES map contains the known remote locations for the source
-   code needed to build Envoy.
+"""The KNOWN_REPOSITORIES map contains the known remote locations for the
+   source code needed to build Envoy and NightHawk
 """
-_KNOWN_REPOSITORIES = {'envoy': 'https://github.com/envoyproxy/envoy.git'}
+_KNOWN_REPOSITORIES = {
+    proto_source.SourceRepository.SourceIdentity.SRCID_ENVOY: \
+      'https://github.com/envoyproxy/envoy.git',
+    proto_source.SourceRepository.SourceIdentity.SRCID_NIGHTHAWK: \
+      'https://github.com/envoyproxy/nighthawk.git'
+}
 
 # In the list of benchmarks to execute, the Baseline image is always
 # the last entry
 BASELINE = -1
 
+def _extract_tag_from_image(image_name: str) -> str:
+  """Pull the tag from the docker image name.
+
+  If there is no tag detected, we return the passed in image name
+
+  Args:
+    image_name: The docker image name
+
+  Return:
+    a string containing the image tag. For example:
+      envoyproxy/envoy:v1.15.3 -> v1.15.3
+  """
+  if ':' not in image_name:
+    return image_name
```

Comment:
We already have an excellent line coverage (above 99%). Normally that would be more than enough, however the two uncovered lines here seem to represent interesting behaviors. Sadly bugs usually hide in uncovered lines.

Would it be feasible to add test case to cover this line? Or remove the code if the behavior isn't useful. I will mark the other uncovered line also.

Reply:
Might be an easier path removing these lines.  I added them to avoid an `IndexError` if the split fails